src/main/java/uk/gov/hmcts/reform/idam/health/vault/VaultEnvironmentPostProcessor.java

```diff
@@ -36,27 +37,36 @@
             "appinsights-instrumentationkey", "azure.application-insights.instrumentation-key"
     );
 
-    private KeyVaultClientProvider keyVaultClientProvider;
+    private final KeyVaultClientProvider provider;
 
     public VaultEnvironmentPostProcessor() {
-        this((clientId, clientKey) -> {
-            KeyVaultCredentials credentials = new ClientSecretKeyVaultCredential(clientId, clientKey);
-            return new KeyVaultClient(credentials);
+        this(new KeyVaultClientProvider() {
+            @Override
+            public KeyVaultClient getClient(String credentialType, String clientId, String clientKey) {
+                if (ACCESS_TOKEN_TYPE.equals(credentialType)) {
```

Comment:
Might consider moving this `if` to a AzureKeyVaultCredentialsFactory` class (interface?).